src/Aspire.Hosting.Azure/AzureOpenAIResource.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.ObjectModel;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an Azure OpenAI resource.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class AzureOpenAIResource(string name) : Resource(name), IAzureResource, IResourceWithConnectionString
+{
+    private readonly Collection<AzureOpenDeploymentResource> _deployments = [];
```

Comment:
If the children already implement `IResourceWithParent<T>` then the `AzureOpenAIResource` might not necessarily need to keep track of its children. In fact if someone then went into the resource graph and removed the deployment resource this state would now be out of sync (granted thats not something that is common).